lib/OpenQA/Base.pm

```diff
@@ -13,20 +13,110 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# This package contains shared functions between WebAPI and WebSockets
-package OpenQA::ServerStartup;
-
+package OpenQA::Base;
+use Mojo::Log;
+use Mojo::Home;
 use strict;
 use warnings;
+use Mojo::Base -base;
+use Sys::Hostname;
+use File::Spec::Functions 'catfile';
+use Mojo::File 'path';
 use Config::IniFiles;
 use db_profiler;
 use db_helpers;
 use OpenQA::Utils;
-use Mojo::File 'path';
 
-sub read_config {
-    my $app = shift;
+has config => sub { {} };
+
+has log => sub { setup_log(shift) };
```

Comment:
Have this set up as `has log;` instead to avoid  perl (and us) getting confused of where the setup is being done, when called out of context.

Reply:
Also I think this is the culprit for those unexpected code changes